### PR TITLE
[Dataset] SSTDataset

### PR DESCRIPTION
**Pull request by HuXiangkun**
## Checklist
Please feel free to remove inapplicable items for your PR.
- [x] The PR title starts with [$CATEGORY] (such as [NN], [Model], [Doc], [Feature]])
- [x] Changes are complete (i.e. I finished coding on this PR)
- [ ] All changes have test coverage
- [x] Code is well-documented
- [ ] To the my best knowledge, examples are either not affected by this change,
      or have been fixed to be compatible with this change
- [ ] Related issue is referred in this PR

### CODE REVIEW

File: examples/pytorch/tree_lstm/train.py [hunk 1]
```diff
@@ -9,7 +9,7 @@
 from torch.utils.data import DataLoader
 
 import dgl
-from dgl.data.tree import SST, SSTBatch
+from dgl.data.tree import SSTDataset, SSTBatch
```
Comment (hetong007) [inline examples/pytorch/tree_lstm/train.py:12]: Seems we are not using the imported `SSTBatch`

File: python/dgl/data/tree.py [hunk 2]
```diff
@@ -125,6 +171,49 @@ def _rec_build(nid, node):
...
+            self._vocab = info['vocab']
+            self._pretrained_emb = info['embed']
+
+    @property
+    def trees(self):
+        return self._trees
+
+    @property
+    def vocab(self):
+        r"""
```
Comment (hetong007) [inline python/dgl/data/tree.py:204]: I'm still unclear about it: to which level do we need the docstrings? We might need an offline discussion

File: python/dgl/data/tree.py [hunk 3]
```diff
@@ -43,70 +41,118 @@ class SST(object):
...
+            - ndata['x']: word id of the node
+            - ndata['y']: label of the node
+            - ndata['mask']: 1 if the node is a leaf, otherwise 0
+        num_classes : int, number of classes for each node
+        pretrained_emb: float tensor, pretrained glove embedding with respect the vocabulary.
+        vocab_size : int, the size of the vocabulary
+
+    Examples
+    --------
+    >>> data = SSTDataset()
```
Comment (classicsong) [inline python/dgl/data/tree.py:71]: Do we need to show how to get train dataset, val dataset and test dataset.?

File: python/dgl/data/tree.py [hunk 4]
```diff
@@ -34,6 +30,8 @@ class SST(object):
     negative, neutral, positive and very positive). The sentiment label is a
     int value stored in the ``y`` feature field.
 
+    Official site: http://nlp.stanford.edu/sentiment/index.html
```
Comment (classicsong) [inline python/dgl/data/tree.py:33]: Any statics for this dataset?

**Comment by hetong007** [comment]
LGTM if passes CI.

**Pull request merged**